test/catch2/live_restore/live_restore_test_env.cpp

```diff
@@ -0,0 +1,63 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "live_restore_test_env.h"
+
+#include <fstream>
+namespace utils {
+
+/*
+ * This class sets up and tears down the testing environment for Live Restore. It spins up a normal
+ * WiredTiger database and then removes all content to leave an empty destination and source folder.
+ * Developers are expected to create the respective files in the these folders manually.
+ */
+live_restore_test_env::live_restore_test_env()
+{
+    // Clean up any pre-existing folders. Make sure an empty DB_SOURCE exists
+    // as it need to exist to open the connection in live restore mode.
+    testutil_remove(_DB_DEST.c_str());
+    testutil_remove(_DB_TEMP_BACKUP.c_str());
+    testutil_recreate_dir(_DB_SOURCE.c_str());
+
+    /*
+     * We're using a connection to set up the file system and let us print WT traces, but all of our
+     * tests will use empty folders where we create files manually. The issue here is
+     * wiredtiger_open will create metadata and turtle files on open and think it needs to remove
+     * them on close. Move these files to a temp location. We'll restore them in destructor before
+     * _conn->close() is called.
+     */
+    static std::string cfg_string = "live_restore=(enabled=true, path=" + _DB_SOURCE + ")";
+    _conn = std::make_unique<connection_wrapper>(_DB_DEST.c_str(), cfg_string.c_str());
```

Comment:
If we have a `connection_wrapper` can't we just create the file system ourselves by calling the `__wt_os_live_restore_fs` directly? Though actually I don't see how this is using a true WT connection?

Reply:
I tried it initially but we segfault because the underlying `__posix_open_file` call assumes that `conn->log_mgr` is non-NULL. I figured I'd use a real connection rather than chasing potentially multiple issues like this with a mock.

I'll have a timeboxed attempt today to move back to the mock connection with additional NULL checks in the posix layer and see how that goes.

Reply:
I'll spin out a separate tech debt ticket to move `live_restore_test_env.h` to using a mock connection instead of a connection wrapper.